parse include in ssh config, fix: json provider, signer in executor

elv-gilles opened the pull request:
* ssh-config: parse includes
* config: fix setting user: set to default only if not provided by config blocks
* fix json provider (broken at 4c08ffe8)
* fix signers in executor

seveas commented:
Thanks!

I've grabbed the json provider fix in a separate commit. For the other changes, I'd like to see separate pr's with more descriptive commits. For instance, you don't explain what is fixed with the signers in the executor, or  why setting the user to default is bad.

elv-gilles commented:
> why setting the user to default is bad

It is not bad, just misplaced: in function `forHost`, the call to `b.updateFromBlock` won't update the value of the User field if it's already set. This is correct given the way blocks are traversed, but it must not be set to the default user before traversing the blocks (but only after in case it's not set). The unit tests were showing that with a config using includes. 
I've added one with a simple config. 

> what is fixed with the signers in the executor

I thought the comment I added in the code was enough. The problem is that the `*agent.signersByPath` is filled with a key that comes from the comment field of the signers from the actual ssh-agent and assumes this is a file path - which is not always true. 

for example:
```
$ ssh-add -l
256 SHA256:UxzdRwuFODt+7iXLv6kRcKZTd5wdZSbxsLqHlrJ4WeI  (ED25519)
256 SHA256:UxzdRwuFODt+7iXLv6kRcKZTd5wdZSbxsLqHlrJ4WeI  (ED25519-CERT)
```

Hence if the config has an IdentityFile not in the signersByPath, the current code returns an empty `[]ssh.Signer` which makes it certain to fail downstream.

Actually - thinking at it - the function `*agent.SignersForPath` should just defer to `a.signers` when the path is not found.
This is more correct and simplifies the code in executor (added the change to this PR).

> I'd like to see separate pr's with more descriptive commits

I was happy to provide the proposed changes. Please feel free to take them by pieces or in total.
